### ISSUE TITLE
Allow a Target to Have Multiple URIs 

### ISSUE THREAD

**Issue by theory** [theory/sqitch#135]
I noticed that Git allows this. It’s an interesting idea: one might be able to deploy to multiple servers in parallel! Handy when managing lots of identical servers. It would take quite a bit of work to update all the commands that accept targets to notice multiple URIs and fork off processes for each, but not too bad. Would also need to consider what to do if one fails. For example, if you deploy to a target with three URIs, and one fails, should the other two revert?

May not be worthwhile to pursue, as folks can also use`xargs -P` to deploy to multiple targets in parallel.

**Comment by loganbentley** [comment]
Has this been brought up again recently? Sqitch looks like a great solution for database change management but we need a solution that can deploy in parallel.

**Comment by theory** [comment]
No, not recently, and I haven't had the chance to work on Sqitch for a while. Contributions warmly welcomed, though.

**Comment by ra1fee** [comment]
How can I formally edit the URI of a project that has already been created. I modified the URI in my sqitch.plan, and now when I try to deploy I get the follwing; 
`Cannot register "schema_name" with URI https://github.com/Attribytes1/adi-etl.git/tree/master/schema_name/sql/: already exists with URI https://github.com/Attribytes/adi-etl/tree/feature/schema_name_onboarding/schema_name/sql/`

**Comment by ra1fee** [comment]
Created a wiki to address my question.[( https://github.com/sqitchers/sqitch/wiki/How-to-edit-URI-on-an-existing-project)]

**Comment by theory** [comment]
Yeah, that would be the way to update it if you wanted to change it, assuming you haven't deployed it to to a lot of locations that would be a PITA to update. Changing namespaces is hard once code ships!

**Comment by chris-bor** [comment]
> Created a wiki to address my question.https://github.com/sqitchers/sqitch/wiki/How-to-edit-URI-on-an-existing-project

Step 5 is good? I get result: "Unknown argument "project_name".

**Comment by theory** [comment]
Oh, right, the URI is included in the hash for changes, isn't it? You'd have to use a [`sqitch plan`](https://sqitch.org/docs/manual/sqitch-plan/) format to get a list of changes and their SHA1 hashes then update all the relevant records in the `changes` table. `tags` too probably.

**Comment by chris-bor** [comment]
That's right. Where I can find info how to calculate new hashes?

**Comment by theory** [comment]
Maybe do this:

```
sqitch plan -f 'format:%H %n' > before.txt
```

Then change the URI in the plan file and run it again:

```
sqitch plan -f 'format:%H %n' > after.txt
```

Then use the contents of the before file for the old IDs, and the after file for the new IDs.

But are you really sure you want to do this? The URI is used only for extra uniqueness. It's never used for linking to a site or project or anything, at least not by Sqitch itself.
